qiskit/qobj/_validation.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2018, IBM.
+#
+# This source code is licensed under the Apache License, Version 2.0 found in
+# the LICENSE.txt file in the root directory of this source tree.
+
+"""QObj validation module for validation against JSON schemas."""
+
+import json
+import os
+import logging
+import jsonschema
+
+from qiskit import QISKitError
+from qiskit import __path__ as qiskit_path
+
+_SCHEMAS = {}
+
+
+logger = logging.getLogger(__name__)
+
+
+def _load_qobj_schema():
+    """Loads the QObj schema for use in future validations."""
+    if 'qobj' not in _SCHEMAS:
+        sdk = qiskit_path[0]
+        # TODO: `transitional_qobj_schema.json` should be replaced with
+        # `qobj_schema.json` once we can ensure Qiskit unrollers are emitting
+        # valid Qobj structures and the Json Schema extension mechanism is in
+        # place.
+        schema_file_path = os.path.join(sdk, 'schemas', 'transitional_qobj_schema.json')
+        with open(schema_file_path, 'r') as schema_file:
+            _SCHEMAS['qobj'] = json.load(schema_file)
+
+    return _SCHEMAS['qobj']
+
+
+def validate_qobj_against_schema(qobj):
+    """Validates a QObj against a schema."""
+    qobj_schema = _load_qobj_schema()
+
+    try:
+        jsonschema.validate(qobj.as_dict(), qobj_schema)
```

Comment:
By default `jsonschema.validate` will validate the schema. It would be better to create a new method 
```
def create_validator(schema, check_schema=True,
                     validator_class=None, *args, **kwargs):
    """
    Generate validator for JSON schema.

    Parameters
    ----------
    schema (dict): JSON schema `dict`.

    check_schema (bool): Verify schema is valid.

    validator_class (jsonschema.IValidator): JsonSchema IValidator instance. Default
            behavior is to determine this from the schema `$schema` field.

    Other positional and keywork arguments will be passed onto `jsonschema`
    validator.
    Returns
    -------
    validator (qutip.IValidator): Validator for JSON schema.
    """



    if not isinstance(schema, dict):
        raise ValueError('''Supplied schema must be a Python dictionary
                            with proper schema form.
                         ''')
    #Resolve Json spec from schema if needed
    if validator_class is None:
        validator_class = jsch.validators.validator_for(schema)

    #verify supplied schema is valid
    if check_schema:
        validator_class.check_schema(schema)

    # generate and return qobj validator
    return validator_class(schema, *args, **kwargs)
```

This will return a validator that may be called as `validator.validate`. This also had the nice feature that we may call `create_validator(...,check_schema=True) in our test suite against all schemas to verify that they are compliant. This work would then not need to be repeated for every validation call.